src/common/Configuration/Config.cpp

```diff
@@ -120,7 +216,22 @@ T ConfigMgr::GetValueDefault(std::string const& name, T def, bool quiet) const
     }
     catch (bpt::ptree_bad_path const&)
     {
-        if (!quiet)
+        char *envVar = EnvVarForIniKey(name);
+        if (envVar)
+        {
+            std::string value = std::string(envVar);
+            
+            {
+                std::lock_guard<std::mutex> lock(_configLock);
```

Comment:
Do we also need to add lock during reading from `_config`?

Reply:
I wonder if it's better to just return envVar without updating _config. 
The code already prints a warning if a config is missing, so ideally one should update their configs if there is something missing. Usually one caches the return value of GetValueDefault(), either in a local variable or in m_int_configs[], so whatever performance slowdown happens with calling EnvVarForIniKey() should be negligible and allow to keep the implementiation more simple, without spreading locks everywhere.

Reply:
@jackpoz Yeah, in that case it's better to make as you suggested.

So it should look like that, right?
```
catch (bpt::ptree_bad_path const&)
    {
        char *envVar = EnvVarForIniKey(name);
        if (envVar)
        {
            if (!quiet)
                TC_LOG_WARN("server.loading", "Missing name %s in config file %s, recovered with environment '%s' value.", name.c_str(), _filename.c_str(), envVar);

            return boost::lexical_cast<T>(envVar);
        }
        else if (!quiet)
        {
            TC_LOG_WARN("server.loading", "Missing name %s in config file %s, add \"%s = %s\" to this file",
                name.c_str(), _filename.c_str(), name.c_str(), std::to_string(def).c_str());
        }
    }
```

I'm not sure how it's better to cast string to T type, came up only with `boost::lexical_cast` solution.

Reply:
Looks good, no idea about the string to T either

Reply:
We have `Trinity::StringTo<T>` in [`StringConvert.h` ](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/common/Utilities/StringConvert.h)for that.